### ISSUE TITLE
`serviceEndpoint` not set when not providing `mediatorConnectionsInvite` to `InitConfig`

### ISSUE THREAD

**Issue by karimStekelenburg** [hyperledger/aries-framework-javascript#401]
This issue is probably related to #381 

When I start my agent for the first time and pass the `mediatorConnectionsInvite` to the `InitConfig`, it successfully connects to the mediator and everything works fine. However, if I restart my agent with exactly the same `InitConfig` configuration, my application hangs at the `agent.initialize()` call.

I can resolve this by removing the `mediatorConnectionsInvite` param from the `InitConfig`, this will make the agent initialize successfully and reconnect to the mediator. However, because I'm not passing `mediatorConnectionsInvite` to the `InitConfig`, `serviceEndpoint` in the invitation that is returned from `agent.connections.createConnection()` is not set to the mediator url. 

In other words: I can only get a valid invitation object at the first run. Thereafter any agent I pass the invitation object to won't know how to reach my agent.

In my view the agent should start without problems when `mediatorConnectionsInvite` is set, regardless of any existing mediator connections. I think @TimoGlastra's suggestion in #381 makes sense. Furthermore, I think the `serviceEndpoint` value inside the `invitation` shouldn't depend on the `InitConfig`, but should instead be based on a `MediatorRecord` (if any).

**Comment by TimoGlastra** [comment]
@burdettadam could you take a look at this?

**Comment by burdettadam** [comment]
Yes, I will take a look at this. I think I see the problem.

**Comment by TimoGlastra** [comment]
Fixed in #419
